odml/dtypes.py

```diff
@@ -178,43 +180,50 @@ def date_get(string):
 
 
 def datetime_get(string):
-    if not string: return None
+    if not string:
+        return None
     if type(string) is datetime.datetime:
-        return string.strftime('%Y-%m-%d %H:%M:%S')
+        return datetime.datetime.strptime(string.strftime('%Y-%m-%d %H:%M:%S'), '%Y-%m-%d %H:%M:%S')
     else:
         return datetime.datetime.strptime(string, '%Y-%m-%d %H:%M:%S')
 
 
 def datetime_set(value):
-    if not value: return None
+    if not value:
+        return None
     if type(value) is datetime.datetime:
         return value.strftime('%Y-%m-%d %H:%M:%S')
     else:
         return datetime.datetime.strptime(value, '%Y-%m-%d %H:%M:%S')
 
 
 def boolean_get(string):
-    if not string: return None
+    if string is None:
+        return None
     if type(string) is bool:
         string = str(string)
     string = string.lower()
     truth = ["true", "t", "1"] # be kind, spec only accepts True / False
-    if string in truth: return True
+    if string in truth:
+        return True
     false = ["false", "f", "0"]
-    if string in false: return False
+    if string in false:
+        return False
     raise ValueError("Cannot interpret '%s' as boolean" % string)
 
 
 def boolean_set(value):
-    if value is None: return None
+    if value is None:
+        return None
     return str(value)
 
 
 def tuple_get(string, count=None):
     """
     parse a tuple string like "(1024;768)" and return strings of the elements
     """
-    if not string: return None
+    if not string:
+        return None
     string = string.strip()
     assert string.startswith("(") and string.endswith(")")
```

Comment:
don't know, in the case string is None, nothing will happen, when we throw an exception, the application will break (if not caught, of course). I would rather not throw an exception.